hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/DefaultSource.scala

```diff
@@ -120,7 +120,7 @@ class DefaultSource extends RelationProvider
              (MERGE_ON_READ, QUERY_TYPE_READ_OPTIMIZED_OPT_VAL, false) =>
           resolveBaseFileOnlyRelation(sqlContext, globPaths, userSchema, metaClient, parameters)
         case (COPY_ON_WRITE, QUERY_TYPE_INCREMENTAL_OPT_VAL, _) =>
-          new IncrementalRelation(sqlContext, parameters, userSchema, metaClient)
+          new CopyOnWriteIncrementalRelation(sqlContext, parameters, userSchema, metaClient)
```

Comment:
if we want to land this for 0.12, we should have a config to guard, otherwise risky